ada/language/ast.py

```diff
@@ -407,7 +385,7 @@ def logic_val(from_node=T.AdaNode.entity, lvar=LogicVar,
         success = Var(If(
             try_immediate & Not(lvar.get_value.is_null),
             True,
-            from_node.parents.find(lambda p: p.xref_entry_point).resolve_names
+            from_node.resolve_names_from_closest_entry_point
```

Comment:
Since you're at that, p_xref_entry_point doesn't follow RM terminology.

The ARM calls what we call an xref_entry_point a "complete context for name resolution":

> 4   Overload resolution is applied separately to each complete context, not
> counting inner complete contexts. Each of the following constructs is a
> complete context:
> 
> 5     * A context_item.
> 
> 6     * A declarative_item or declaration.
> 
> 7     * A statement.
> 
> 8     * A pragma_argument_association.
> 
> 9     * The expression of a case_statement.

I wonder if we shouldn't change terminology to fit the RM here